src/Microsoft.DotNet.Interactive.CSharpProject/RoslynWorkspaceUtilities/RoslynWorkspaceUtilities.cs

```diff
@@ -0,0 +1,332 @@
+                    filePath: x)).ToList();
+
+    private static IReadOnlyList<ProjectReference> GetExistingProjectReferences(BuildProjectData buildProjectData, CodeAnalysis.Workspace workspace) =>
+        buildProjectData.ProjectReferences
+            .Select(x => workspace.CurrentSolution.Projects.FirstOrDefault(y => y.FilePath.Equals(x, StringComparison.OrdinalIgnoreCase)))
+            .Where(x => x != null)
+            .Select(x => new ProjectReference(x.Id)).ToList();
+
+    private static IReadOnlyList<MetadataReference> GetMetadataReferences(BuildProjectData buildProjectData) =>
+        buildProjectData
+            .References?.Where(File.Exists)
+            .Select(x => MetadataReference.CreateFromFile(x)).ToList();
+
+    private static IReadOnlyList<AnalyzerReference> GetAnalyzerReferences(BuildProjectData buildProjectData, CodeAnalysis.Workspace workspace)
+    {
+        IAnalyzerAssemblyLoader loader = workspace.Services.GetRequiredService<IAnalyzerService>().GetLoader();
+
+        return buildProjectData
+                .AnalyzerReferences?
+                    .Where(File.Exists)
+                    .Select(x => new AnalyzerFileReference(x, loader)).ToList();
+    }
+
+    private static ParseOptions CreateParseOptions(BuildProjectData buildProjectData, string languageName)
+    {
+        if (languageName == LanguageNames.CSharp)
+        {
+            return CreateCSharpParseOptions(buildProjectData);
+        }
+
+        return null;
+    }
+
+    private static CSharpParseOptions CreateCSharpParseOptions(BuildProjectData buildProjectData)
+    {
+        CSharpParseOptions parseOptions = new CSharpParseOptions();
+
+        parseOptions = parseOptions.WithPreprocessorSymbols(buildProjectData.PreprocessorSymbols);
+
+        string langVersion = buildProjectData.LangVersion;
+        if (!string.IsNullOrWhiteSpace(langVersion)
+            && LanguageVersionFacts.TryParse(langVersion, out LanguageVersion languageVersion))
+        {
+            parseOptions = parseOptions.WithLanguageVersion(languageVersion);
+        }
+
+        return parseOptions;
+    }
+
+    private static CompilationOptions CreateCompilationOptions(BuildProjectData buildProjectData, string languageName)
+    {
+        string outputType = buildProjectData.OutputType;
+        OutputKind? kind = null;
+        switch (outputType)
+        {
+            case "Library":
+                kind = OutputKind.DynamicallyLinkedLibrary;
+                break;
+            case "Exe":
+                kind = OutputKind.ConsoleApplication;
+                break;
+            case "Module":
+                kind = OutputKind.NetModule;
+                break;
+            case "Winexe":
+                kind = OutputKind.WindowsApplication;
+                break;
+        }
+
+        if (kind.HasValue)
+        {
+            if (languageName == LanguageNames.CSharp)
```

Comment:
Same comment. And removing this generalization here means we avoid a code branch where this can return null.

Reply:
done

Reply:
fixed